src/stores/challenge.ts

```diff
@@ -8,32 +9,88 @@ export enum ChallengeStatus {
   after = 'after',
 }
 
+// fixtures
+import thisCampaignFixture from '../../test/cypress/fixtures/thisCampaign.json';
+
 // types
 import type { Logger } from '../components/types/Logger';
+import type { Phase } from '../components/types/Challenge';
+
+const phaseSet = thisCampaignFixture.results[0].phase_set as Phase[] | null;
 
 export const useChallengeStore = defineStore('challenge', {
   state: () => ({
     // property set in pinia.js boot file
     $log: null as Logger | null,
-    isChallengeActive: false,
-    challengeStatus: ChallengeStatus.before,
+    isChallengeActive: true,
+    /**
+     * Phase set for the current campaign
+     * Phase object with id `registration` marks the ability to register.
+     * Phase object with id `competition` marks the duration of the challenge.
+     * Phase object with id `entry_enabled` marks the ability to log routes.
+     * Phase object with id `payment` marks the ability to pay.
+     * Phase object with id `invoices` marks the ability to see invoices.
+     */
+    phaseSet: phaseSet ? phaseSet : [],
   }),
 
   getters: {
-    getIsChallengeActive: (state): boolean => state.isChallengeActive,
-    getChallengeStatus: (state): ChallengeStatus => state.challengeStatus,
-  },
-
-  actions: {
-    setIsChallengeActive(isActive: boolean): void {
-      this.isChallengeActive = isActive;
+    getChallengeStatus: (): ChallengeStatus => {
+      const thisStore = useChallengeStore();
+      if (thisStore.getIsChallengeInActivePhase) {
+        return ChallengeStatus.during;
+      } else if (thisStore.getIsChallengeInRegistrationPhase) {
+        return ChallengeStatus.before;
+      }
+      return ChallengeStatus.after;
+    },
+    /**
+     * Gets active challenge status
+     * Status is based on phase_set array variable.
+     * @returns {boolean}
+     */
+    getIsChallengeInActivePhase: (store): boolean => {
+      const competitionPhase = store.phaseSet.find(
+        (phase: Phase) => phase.phase_type === PhaseType.competition,
+      );
+      if (competitionPhase) {
```

Comment:
Lines 56-69 has same logic as lines 71-87, make it as standalone function, please.